pkg/controllers/rollout/controller.go

```diff
@@ -362,40 +366,39 @@ func (r *Reconciler) pickBindingsToRoll(ctx context.Context, allBindings []*flee
 				// it is being deleted, it can be removed from the cluster at any time, so it can be unavailable at any time
 				canBeUnavailableBindings = append(canBeUnavailableBindings, binding)
 			}
-
 		case fleetv1beta1.BindingStateScheduled:
 			// the scheduler has picked a cluster for this binding
 			schedulerTargetedBinds = append(schedulerTargetedBinds, binding)
-			// this binding has not been bound yet, so it is an update candidate
-			// pickFromResourceMatchedOverridesForTargetCluster always returns the ordered list of the overrides.
-			cro, ro, err := r.pickFromResourceMatchedOverridesForTargetCluster(ctx, binding, matchedCROs, matchedROs)
+			boundingCandidate, err := r.processScheduledBindings(ctx, binding, crp, latestResourceSnapshot, matchedCROs, matchedROs)
 			if err != nil {
-				return nil, nil, false, err
+				return nil, nil, false, 0, err
+			}
+			if boundingCandidate.currentBinding != nil {
```

Comment:
I don't see much benefit of this refactor, the function is not reused and it didn't shorten the lines of code either.